pkg/filter/filter_test.go

```diff
@@ -0,0 +1,345 @@
+			obj: func() *unstructured.Unstructured {
+				uo := &unstructured.Unstructured{}
+				uo.SetLabels(map[string]string{"some-label": "/apis/sro.openshift.io/v1"})
+
+				return uo
+			}(),
+			matcher: BeTrue(),
+		},
+		{
+			name:    "no selflink",
+			obj:     &unstructured.Unstructured{},
+			matcher: BeFalse(),
+		},
+	}
+
+	entries := make([]TableEntry, 0, len(cases))
+
+	for _, c := range cases {
+		entries = append(entries, Entry(c.name, c.obj, c.matcher))
+	}
+
+	DescribeTable(
+		"should return the correct value",
+		func(obj client.Object, m types.GomegaMatcher) {
+			Expect(filter.IsSpecialResource(obj)).To(m)
+		},
+		entries...)
+})
+
+var _ = Describe("Owned", func() {
+	cases := []struct {
+		name    string
+		obj     client.Object
+		matcher types.GomegaMatcher
+	}{
+		{
+			name: "via ownerReferences",
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					OwnerReferences: []metav1.OwnerReference{
+						{Kind: "SpecialResource"},
+					},
+				},
+			},
+			matcher: BeTrue(),
+		},
+		{
+			name: "via labels",
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"specialresource.openshift.io/owned": "whatever"},
+				},
+			},
+			matcher: BeTrue(),
+		},
+		{
+			name:    "not owned",
+			obj:     &corev1.Pod{},
+			matcher: BeFalse(),
+		},
+	}
+
+	entries := make([]TableEntry, 0, len(cases))
+
+	for _, c := range cases {
+		entries = append(entries, Entry(c.name, c.obj, c.matcher))
+	}
+
+	DescribeTable(
+		"should return the expected value",
+		func(obj client.Object, m types.GomegaMatcher) {
+			Expect(filter.Owned(obj)).To(m)
+		},
+		entries...,
+	)
+})
+
+var _ = Describe("Predicate", func() {
+	resetMode := func() { filter.Mode = "" }
+
+	AfterEach(resetMode)
+
+	Context("CreateFunc", func() {
+		cases := []struct {
+			name       string
+			obj        client.Object
+			retMatcher types.GomegaMatcher
+		}{
+			{
+				name:       "special resource",
+				obj:        &v1beta1.SpecialResource{},
+				retMatcher: BeTrue(),
+			},
+			{
+				name: "owned",
+				obj: &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						OwnerReferences: []metav1.OwnerReference{
+							{Kind: "SpecialResource"},
+						},
+					},
+				},
+				retMatcher: BeTrue(),
+			},
+			{
+				name:       "random pod",
+				obj:        &corev1.Pod{},
+				retMatcher: BeFalse(),
+			},
+		}
+
+		entries := make([]TableEntry, 0, len(cases))
+
+		for _, c := range cases {
+			entries = append(entries, Entry(c.name, c.obj, c.retMatcher))
+		}
+
+		DescribeTable(
+			"should work as expected",
+			func(obj client.Object, m types.GomegaMatcher) {
+				ret := filter.Predicate().Create(event.CreateEvent{Object: obj})
+
+				Expect(ret).To(m)
+				Expect(filter.Mode).To(Equal("CREATE"))
+			},
+			entries...,
+		)
+	})
+
+	Context("UpdateFunc", func() {
+		It("should work as expected", func() {
+			Skip("Testing this function requires injecting a fake ClientSet")
+		})
+	})
+
+	Context("DeleteFunc", func() {
```

Comment:
Same as https://github.com/openshift-psap/special-resource-operator/pull/154/files#r761802431.

We should only test that `Predicate` return what it is expected to return by using mocks on `Delete`

Reply:
`DeleteFunc` is not defined outside of `Predicate`, nor is it injected. I don't know how we could compare that `Predicate().Delete()` returns `DeleteFunc`.  
Besides, `DeleteFunc` is our own code and we need to test it.